src/h3lib/lib/h3Index.c

```diff
@@ -734,3 +760,232 @@ void H3_EXPORT(h3ToGeoBoundary)(H3Index h3, GeoBoundary* gb) {
  *         a Class II grid.
  */
 int isResClassIII(int res) { return res % 2; }
+
+/**
+ * Produces ijk+ coordinates for an index anchored by an origin.
+ *
+ * The coordinate space used by this function may have deleted
+ * regions or warping due to pentagonal distortion.
+ *
+ * Coordinates are only comparable if they come from the same
+ * origin index.
+ *
+ * @param origin An anchoring index for the ijk+ coordinate system.
+ * @param index Index to find the coordinates of
+ * @param out ijk+ coordinates of the index will be placed here on success
+ * @return 0 on success, or another value on failure.
+ */
+int h3ToIjk(H3Index origin, H3Index h3, CoordIJK* out) {
+    if (H3_GET_MODE(origin) != H3_HEXAGON_MODE ||
+        H3_GET_MODE(h3) != H3_HEXAGON_MODE) {
+        // Only hexagon mode is relevant, since we can't
+        // encode directionality in CoordIJK.
+        return 1;
+    }
+
+    int res = H3_GET_RESOLUTION(origin);
+
+    if (res != H3_GET_RESOLUTION(h3)) {
+        return 1;
+    }
+
+    int originBaseCell = H3_GET_BASE_CELL(origin);
+    int baseCell = H3_GET_BASE_CELL(h3);
+
+    // Direction from origin base cell to index base cell
+    int dir = 0;
+    int revDir = 0;
+    if (originBaseCell != baseCell) {
+        for (dir = 1; dir < 7; dir++) {
+            int testBaseCell = _getBaseCellNeighbor(originBaseCell, dir);
```

Comment:
I'd suggest encapsulating this as `Direction getBaseCellNeighborDirection(int origin, int dest)`, in `baseCells.c`, returning `-1` on failure.